Protos/V1/sessions_service.proto

```diff
@@ -0,0 +1,160 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "objects.proto";
+import "session_status.proto";  
+
+option csharp_namespace = "ArmoniK.Api.gRPC.V1.Sessions";
+package ArmoniK.api.grpc.V1.sessions;
+
+/**
+ * The SessionService provides methods for listing, getting and cancelling sessions
+ */
+service Sessions {
+  // Get all sessions with pagination
+  rpc ListSessions(ListSessionsRequest) returns (ListSessionsResponse);
+
+  // Get a session using an ID
+  rpc GetSession(GetSessionRequest) returns (GetSessionResponse) {}
+
+  // Cancel a session using an ID
+  rpc CancelSession(CancelSessionRequest) returns (CancelSessionResponse) {}
+}
+
+/**
+ * A raw session object
+ */
+message SessionRaw {
+  string session_id = 1;
+  SessionStatus.SessionStatus status = 2;
+  repeated string partition_ids = 3;
+  TaskOptions.TaskOptions options = 4;
+
+  google.protobuf.Timestamp created_at = 5;
+  google.protobuf.Timestamp cancelled_at = 6;
+}
+
+/**
+ * A session object
+ */
+message SessionSummary {
+  string session_id = 1;
+
+  int64 count_tasks = 2;
+  int64 count_tasks_pending = 3;
+  int64 count_tasks_error = 4;
+  int64 count_tasks_completed = 5;
+  int64 count_tasks_processing = 6;
+
+  SessionStatus.SessionStatus status = 7;
+  google.protobuf.Timestamp created_at = 8;
+  google.protobuf.Timestamp cancelled_at = 9;
+  google.protobuf.Timestamp last_activity_at = 10;
+}
+
+/**
+ * Request for listing sessions
+ */
+message ListSessionsRequest {
+  // Page number
+  int32 page = 1;
+  // Page size
+  int32 page_size = 2;
+  
+  // To filter returned sessions
+  message Filter {
+    // Name of the application
+    string application_name = 1;
+    // Version of the application
+    string application_version = 2;
+    // Id of the session
+    string session_id = 3;
+    // Date after the last activity of the session
+    google.protobuf.Timestamp last_activity_after = 4;
+    // Date before the last activity of the session
+    google.protobuf.Timestamp last_activity_before = 5;
+    // Date after the creation of the session
+    google.protobuf.Timestamp created_after = 6;
+    // Date before the creation of the session
+    google.protobuf.Timestamp created_before = 7;
+    // Date after the cancellation of the session
+    google.protobuf.Timestamp cancelled_after = 8;
+    // Date before the cancellation of the session
+    google.protobuf.Timestamp cancelled_before = 9;
+  }
+
+  // Filter to apply to the returned sessions
+  Filter filter = 3;
+
+  // Availble fields to order the returned sessions
```

Comment:
Available

Reply:
enum type names should be in capital letters with underscores according to the style guide

Reply:
Okey, will change this